turbo/consumers.py

```diff
@@ -16,39 +15,24 @@ def connect(self):
         self.requests = dict()
         self.accept()
 
-    def notify(self, event, *args, **kwargs):
-        template = event["template"]
+    def notify(self, event,):
         extra_context = event["context"]
-        model_label = event["model"]
-        model = apps.get_model(model_label)
-
-        pk = event["pk"]
         action = event["action"]
-        to_list = event["to_list"]
-
-        if to_list:
-            dom_target = model._meta.verbose_name_plural.lower()
-        else:
-            dom_target = f"{model._meta.verbose_name.lower()}_{pk}"
+        dom_target = event["dom_target"]
 
         for request_id in self.requests[event["channel_name"]]:
-            instance = model.objects.get(pk=pk)
```

Comment:
I'm confused about how the template is rendered for model streams without this line / adding the instance to the template context before rendering. Since models can't be serialized into the channel layer, we do need some way to recover the instance itself

Reply:
It already is in the context Line 19. I agree with your statement that we have to thing about de-/serialisation. But that brings me to the point where I woudl ask if it doenst make sense to only send rendered templates over to the notify method. Then we get rid of all serialization issues.

The reason why I had to remove it here is that we have an Instance in the Model Szenario but not in the "every other message" scenario. So this is an "implementation detail" of the BroadcastModelMixin and not of the Consumer IMHO.